Automattic/jetpack
# Extend media validation to videos

This change extends the `use-media-restrictions` hook, now it can validate videos too.

#### Changes proposed in this Pull Request:

* Added video validation
* Refactored `media-section` a bit, now it always selects the media, which is then validated in the `useEffect` hook. This solution seems cleaner, and we also get rid of parsing media data from 2 different places.
* Changed `attached_media` so we send the URL to WPCOM too, to avoid using different images because of scaling

#### Other information:


#### Jetpack product discussion



1203495687561698-as-1203420392934656

#### Does this pull request change what data or activity we track or use?



No.

#### Testing instructions:






* Image validation should work as before, use #27610 testing instruction to test that.
* Try uploading a video, it should get selected.
* Try uploading a video that is over the size limit - you can tweak the limits in `RESTRICTIONS` to test this.
* Try uploading a video that is over the length limit.

This is a neat site for test videos: https://www.sample-videos.com/

Run tests with:
`cd projects/js-packages/publicize-components/` then `npm test`
